Extension/src/LanguageServer/configurations.ts

```diff
@@ -386,6 +395,46 @@ export class CppProperties {
         }
     }
 
+    private nodeAddonMap: {[dependency: string]: string} = {
+      "nan": "node --no-warnings -e \"require('nan')\"",
+      "node-addon-api": "node --no-warnings -p \"require('node-addon-api').include\"",
+    };
+
+    private async addNodeAddonIncludeLocations(rootPath: string) : Promise<void> {
+        try {
+
+            const settings: CppSettings = new CppSettings(this.rootUri);
+            if (settings.addNodeAddonIncludePaths) {
+                const package_json = JSON.parse(await pfs.readFile(path.join(rootPath, "package.json"), "utf8"));
+                dependencyLoop:
+                for (const dep in this.nodeAddonMap) {
+                  if (dep in package_json.dependencies) {
+                      const execCmd: string = this.nodeAddonMap[dep];
+                      let stdout = await util.execChildProcess(execCmd, rootPath);
+                      if (!stdout) continue dependencyLoop;
+
+                      // cleanup newlines
+                      if (stdout[stdout.length - 1] === "\n") {
+                          stdout = stdout.slice(0, -1);
+                      }
+                      // node-addon-api returns a quoted string, e.g., '"/home/user/dir/node_modules/node-addon-api"'.
+                      if (stdout[0] === "\"" && stdout[stdout.length - 1] === "\"") {
+                          stdout = stdout.slice(1, -1);
+                      }
+                      if (stdout) {
+                          this.nodeAddonIncludes.push(stdout);
```

Comment:
I think you should check if the stdout is a valid directory, maybe calling util.checkDirectoryExists. If it doesn't exist, you could check if path.join("${workspaceFolder}", stdout), since I'm getting back a relative path `node_modules\\nan`.

Reply:
nan always returns the path relative to the working directory, so i'm calling `path.resolve()` on the results and checking that. i think the only time `util.checkDirectoryExists()` will fail is if the node command executed by `addNodeAddonIncludeLocations()` fails.

Reply:
path.resolve doesn't work because it uses a working path of VS Code, such as `C:\\Users\\<user>\\AppData\\Local\\Programs\\Microsoft VS Code\`, and not the user's workspace. Was there a reason `path.join("${workspaceFolder}", stdout)` or something similar wouldn't work?

Reply:
no reason that doesn't work. i was trying to get it resolved with one shot as opposed to looking around but obviously hadn't checked my assumption about the working directory. i will address this in the next push.